system/Config/DotEnv.php

```diff
@@ -76,34 +76,24 @@ public function __construct(string $path, string $file = '.env')
 	 */
 	public function load(): bool
 	{
-		$vars = $this->parse();
+		$loaded = $this->parse();
 
-		if ($vars === null)
-		{
-			return false;
-		}
-
-		foreach ($vars as $name => $value)
-		{
-			$this->setVariable($name, $value);
-		}
-
-		return true; // for success
+		return $loaded; // for success
 	}
 
 	//--------------------------------------------------------------------
 
 	/**
 	 * Parse the .env file into an array of key => value
 	 *
-	 * @return array|null
+	 * @return boolean
 	 */
-	public function parse(): ?array
+	public function parse(): bool
```

Comment:
change return type of public method is always bc break

Reply:
Agree with this, we need a way to make this fix work without changing the return type.